src/client/options/mod.rs

```diff
@@ -675,43 +675,181 @@ impl Serialize for ClientOptions {
+    /// autodiscover all servers in the cluster.
+    ///
+    /// The default value is false.
     pub direct_connection: Option<bool>,
+
+    /// The credential to use for authenticating connections made by this client.
     pub credential: Option<Credential>,
+
+    /// Default database for this client.
+    ///
+    /// By default, no default database is specified.
     pub default_database: Option<String>,
-    max_staleness: Option<Duration>,
+
+    /// Whether or not the client is connecting to a MongoDB cluster through a load balancer.
+    pub load_balanced: Option<bool>,
+
+    /// Amount of time spent attempting to send or receive on a socket before timing out; note that
+    /// this only applies to application operations, not SDAM.
+    pub socket_timeout: Option<Duration>,
+
+    /// Specifies the level of compression when using zlib to compress wire protocol messages; -1
+    /// signifies the default level, 0 signifies no compression, 1 signifies the fastest speed, and
+    /// 9 signifies the best compression.
+    pub zlib_compression: Option<i32>,
+
+    /// The maximum replication lag, in wall clock time, that a secondary can suffer and still be
+    /// eligible for server selection.
+    pub max_staleness: Option<Duration>,
+
+    /// The authentication mechanism method to use for connection to the server.
+    pub auth_mechanism: Option<AuthMechanism>,
+
+    /// The database that connections should authenticate against.
+    pub auth_source: Option<String>,
+
+    /// Additional options provided for authentication (e.g. to enable hostname canonicalization
+    /// for GSSAPI).
+    pub auth_mechanism_properties: Option<Document>,
+
+    /// Default read preference for the client (excluding tags).
+    pub read_preference: Option<ReadPreference>,
+
+    /// Default read preference tags for the client; only valid if the read preference mode is not
+    /// primary.
+    ///
+    /// The order of the tag sets in the read preference is the same as the order they are
+    /// specified in the URI.
+    pub read_preference_tags: Option<Vec<TagSet>>,
+
+    pub(crate) srv: bool,
+    wait_queue_timeout: Option<Duration>,
     tls_insecure: Option<bool>,
-    auth_mechanism: Option<AuthMechanism>,
-    auth_source: Option<String>,
-    auth_mechanism_properties: Option<Document>,
-    read_preference: Option<ReadPreference>,
-    read_preference_tags: Option<Vec<TagSet>>,
-    load_balanced: Option<bool>,
     original_uri: String,
 }
 
+impl ConnectionString {
+    /// Amount of time spent attempting to check out a connection from a server's connection pool
+    /// before timing out.  Not supported by the Rust driver.
+    pub fn wait_queue_timeout(&self) -> Option<Duration> {
+        self.wait_queue_timeout
+    }
+
+    /// Relax TLS constraints as much as possible (e.g. allowing invalid certificates or hostname
+    /// mismatches).  Not supported by the Rust driver.
+    pub fn tls_insecure(&self) -> Option<bool> {
```

Comment:
Now that we have OpenSSL support, I _think_ we can / do support this? If not, maybe we should file a ticket about adding support, at least for OpenSSL only.

Reply:
Filed [RUST-1309](https://jira.mongodb.org/browse/RUST-1309) to track this.